src/figuresClasses.ts

```diff
@@ -2,10 +1,75 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: 'triangle' = 'triangle',
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('Every side must be shorter than the sum of others');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const s = Math.sqrt(p * ((p - this.a) * (p - this.b) * (p - this.c)));
+
+    return Math.floor(s * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+    public shape: 'circle' = 'circle',
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('your error message');
```

Comment:
The error message 'your error message' is a placeholder. Consider providing a more descriptive error message, such as 'Radius must be a positive number.'